### DISCUSSION TITLE
Take input from user during execution of command

### DISCUSSION THREAD

**Discussion by mike-5345** [daniele77/cli#204] [Ideas / Feature requests]
Hello Daniele,

For my project, some of my commands need to take input from the user during the execution of a command lambda. Is there a way to do this currently?

If not, would this be something you would be interested in adding?

Thanks,
Mike

**Comment by daniele77** [comment]
Hi Mike.

If you only need local console support, you can use the standard C++ input mechanism (e.g., `std::cin`, `std::getline()`, etc). I never tried, but it should work.

Of course, for remote sessions (i.e., telnet sessions) that mechanism cannot work. You would need a new feature (i.e., the command handler should have a new `std::istream& in` parameter, in addition to the existing one `std::ostream& out`).
This feature is not an easy one to implement, let me know if it's enough local console support or if you need remote sessions too.

**Reply by bohich** [reply]
Hello Daniele,

I use local console with (e.g., std::cin, std::getline(), etc).
But when i write anything to terminal some letter are not recorded. After press enter i get ("wrds"), but wrote ("words").
I can elaborate more on the use case if this question is unclear.
Maybe you know how to fix that.

Thanks

**Reply by daniele77** [reply]
Yes Mike,

actually when using `CliLocalTerminalSession`, you cannot use `std::cin` and `std::getline()` (please see this previous discussion: https://github.com/daniele77/cli/discussions/173 but also this one: https://github.com/daniele77/cli/discussions/222).

If you need to take input from the user inside the command handlers, you can consider using `CliFileSession` instead of `CliLocalTerminalSession` (see the example `simplelocalsession.cpp` that comes with the library). However, in this case, you won't have auto-completion and history.

**Reply by daniele77** [reply]
I've just committed a fix that should enable user keyboard handling on Linux platform as well (commit 17dafd8).
It's the issue #239.
Now you can use `std::cin` and `std::getline` inside command handlers (local session only, of course).
Please, see if this solves your problem.
